Improve SEGMENT_TABLE_DOWNLOAD_COUNT accuracy

rhodo opened the pull request:
SEGMENT_TABLE_DOWNLOAD_COUNT  track how many concurrent segment download is happening at a given time. The current logic only consider concurrency from `_segmentDownloadSemaphore`, but not `_segmentOperationsThrottler`. Move the metrics increment and decrement to very beginning and end right close to segment download.

Also we might not need another atomic counter `_numSegmentsAcquiredDownloadSemaphore` to keep track of everything actually. As the gauge can itself be add or decrement.

Run `OfflineClusterIntegrationTest` and see the gauge being bumped and reset properly.

somandal commented on pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java:
```diff
@@ -962,6 +957,7 @@ protected File downloadSegmentFromDeepStore(SegmentZKMetadata zkMetadata)
                 + "queue-length={}).", segmentName, System.currentTimeMillis() - startTime,
             segmentDownloadThrottler.getQueueLength());
       }
+      _serverMetrics.addValueToTableGauge(_tableNameWithType, ServerGauge.SEGMENT_TABLE_DOWNLOAD_COUNT, 1);
```
`addValueToTableGauge` is deprecated, so we might still need that `AtomicInteger`

rhodo replied:
yea actually I don't know why we deprecate that in the first place, It is common to use inc()/dec() api to with gauge IMO. but keeping _numSegmentsAcquiredDownloadSemaphore also it not a unacceptable burden, I can stick with that.

somandal commented on pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java:
```diff
@@ -997,10 +993,9 @@ protected File downloadSegmentFromDeepStore(SegmentZKMetadata zkMetadata)
         FileUtils.deleteQuietly(tempRootDir);
       }
     } finally {
+      _serverMetrics.addValueToTableGauge(_tableNameWithType, ServerGauge.SEGMENT_TABLE_DOWNLOAD_COUNT, -1);
```
nit: just wondering if it might make sense to decrement after releasing like before? should be fine to keep this as well though

rhodo commented:
@somandal addressed comment, stick with `_numSegmentsAcquiredDownloadSemaphore`, also run `OfflineClusterIntegrationTest` and see the counter being bumped and reset properly.

somandal reviewed (approved):
lgtm! thanks!
